xarray/core/formatting.py

```diff
@@ -145,15 +145,50 @@ def format_array_flat(items_ndarray, max_width):
     return pprint_str
 
 
+def _get_level_values(index, level_num, max_size=50):
+    """Similar to `pd.MultiIndex.get_level_values` but
+    here truncated to `max_size`.
+    """
+    unique = index.levels[level_num]
+    labels = index.labels[level_num]
+    size = min(max_size, labels.size)
+    filled = pd.core.algorithms.take_1d(unique.values, labels[:size],
```

Comment:
In some cases it might make sense to just use `index.levels[level_num][:max_size]` to show the (first) unique values for each level.

But in other cases I find this

```python
>>> data.isel(x=range(3))
<xarray.DataArray (x: 3)>
array([0, 1, 2])
Coordinates:
  * x                    (x) object MultiIndex
    - a_long_level_name  object 'a' 'a' 'a'
    - level_1            int64 0 1 2
```

much better than this:
```python
>>> data.isel(x=range(3))
<xarray.DataArray (x: 3)>
array([0, 1, 2])
Coordinates:
  * x                    (x) object MultiIndex
    - a_long_level_name  object 'a' 'b'
    - level_1            int64 0 1 2 3 4 5 6 7 8 9
```

What about just returning `pd.core.algorithms.take_1d(unique.values, labels[:max_size])` or even `np.take(unique.values, labels[:max_size])`?